### DISCUSSION TITLE
Auto save doesn't trigger when switching helix windows

### DISCUSSION THREAD

**Discussion by iyegoroff** [helix-editor/helix#9642] [Troubleshooting]
I have enabled `auto-save = true` in my `config.toml` and all changes has been saved when switching tmux windows or WM desktops. However, it doesn't save changes when switching helix windows with `Ctrl-w + w`. Am I missing something or this behavior is not supported?

Platform: Linux
Terminal: alacritty 0.13.1
Installation: xbps-install
Helix: 23.10 (f6021dd0)

**Comment by the-mikedavis** [comment]
This is the intended behavior - `auto-save` triggers when the terminal loses focus, or in tmux's case, when the terminal split containing Helix loses focus

**Reply by iyegoroff** [reply]
Ok. Then could you extend auto-saving behavior to trigger on helix window change too?

For example, in VSCode auto-save is defined as a union of string values:

![Screenshot 2024-02-17 at 07 33 01](https://github.com/helix-editor/helix/assets/4447438/69a1856e-2f36-4fe6-ba02-8a457b368d46) ![Screenshot 2024-02-17 at 07 33 21](https://github.com/helix-editor/helix/assets/4447438/0c3f9528-c523-42ab-97ea-c852cb9c6050)

`onWindowChange` triggers saving only when switching from VSCode to another app (like in helix right now)
`onFocusChange` also triggers saving when changing active editor tab inside VSCode itself 

Another possible solution: Adding a command for window mode, so `Ctrl-w` could be remapped to `[":w", "window_mode"]`.

**Reply by the-mikedavis** [reply]
See https://github.com/helix-editor/helix/issues/3451, https://github.com/helix-editor/helix/pull/3178, https://github.com/helix-editor/helix/pull/5303

I think adding additional ways to save automatically would be a nice improvement. In the meantime you could rebind the window switching keys under `C-w` to save and then switch windows.
